modules/renter/repair.go

```diff
@@ -336,6 +330,12 @@ func (r *Renter) managedScheduleChunkRepair(rs *repairState, chunkID chunkID, ch
 		missingPieces = missingPieces[:len(usefulWorkers)]
 	}
 
+	// Erasure code the pieces.
+	pieces, err := file.erasureCode.Encode(chunkData, missingPieces)
+	if err != nil {
+		return build.ExtendErr("unable to erasure code chunk data", err)
+	}
```

Comment:
you'd just add some extra information to the object to indicate whether the upload on that data had completed or not.

I'll pull the locks from the erasure coding package, they don't really seem to be helping memory usage at all.